pkg/broker/broker.go

```diff
@@ -76,23 +82,42 @@ func (br *Broker) ListenForClients() {
 		case sb := <-br.NewClients:
 			br.clients[sb] = true
 			cc, ok := br.clientsCount.Load(sb.ID)
-			if ok {
+			if ok && cc.(int) > 0 {
 				br.clientsCount.Store(sb.ID, cc.(int)+1)
 			} else {
 				br.clientsCount.Store(sb.ID, 1)
+				br.CloseClientC[sb.ID] = make(chan struct{})
+				tl, err := tailer.New(sb.Extra)
+				go func() {
+					br.Log.Printf("[tailer] Starting for %s", sb.ID)
+
+					scanner := bufio.NewScanner(tl)
+					for scanner.Scan() {
+						br.Notifier <- &Event{Msg: []byte(scanner.Text()), ID: sb.ID}
+					}
+				}()
+				go func() {
+					<-br.CloseClientC[sb.ID]
```

Comment:
Sure it would, that way we would notify the Tailer explicitly when it needs to terminate. 

Regardless of how we terminate the Tailer though, either explicitly with a channel message or implicitly with closing the file in question, I think the complexity would remain the same as we need the goroutine in order not to block while listening to the `br.CloseClientC` channel.